### DISCUSSION TITLE
Anyone ever integrated maplibre native into a Dear Imgui app?

### DISCUSSION THREAD

**Discussion by andreamancuso** [ocornut/imgui#7723] [New Users Build/Link/Run issues ONLY!]
Sorry for asking another silly question - I need to embed maps into my application and I came across https://github.com/maplibre/maplibre-native/ I see it supports GLFW so I was wondering if anyone has ever tried integrating this into a Dear Imgui based application.

I've stumbled upon https://github.com/MAPSWorks/webAsmPlay-imgui-osm which apparently uses Dear Imgui (docking branch) and an OpenStreetMap C++ library.

Thank you in advance

**Comment by andreamancuso** [comment]
I have since written simple logic to render a static, raster image based map based on OpenStreetMap data:

![image](https://github.com/user-attachments/assets/99463ff5-684e-45f7-97e3-76cb4ca37c69)

This has helped me understand the basics of map rendering. Next up is rendering of vector based tiles.

**Reply by cjbd** [reply]
hey, i'm thinking something similar, do you mind sharing your code, thanks!

**Reply by andreamancuso** [reply]
This is the repo: https://github.com/andreamancuso/osm-static-map-generator

I am not a C++ developer by trade. The code is suboptimal. I hope it helps nonetheless.

**Reply by cjbd** [reply]
@andreamancuso , thank you!
